src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs

```diff
@@ -261,19 +268,32 @@ public virtual object GetDynamicParameters()
                         // context. Force the client to use that subscription:
                         TemplateSpecsClient.SubscriptionId = resourceIdentifier.Subscription;
                     }
+                    JObject templateObj = (JObject)null;
+                    try
+                    {
+                        var templateSpecVersion = TemplateSpecsClient.TemplateSpecVersions.Get(
+                            ResourceIdUtility.GetResourceGroupName(templateSpecId),
+                            ResourceIdUtility.GetResourceName(templateSpecId).Split('/')[0],
+                            resourceIdentifier.ResourceName);
 
-                    var templateSpecVersion = TemplateSpecsClient.TemplateSpecVersions.Get(
-                        ResourceIdUtility.GetResourceGroupName(templateSpecId),
-                        ResourceIdUtility.GetResourceName(templateSpecId).Split('/')[0],
-                        resourceIdentifier.ResourceName);
-
-                    if (!(templateSpecVersion.Template is JObject))
+                        if (!(templateSpecVersion.Template is JObject))
+                        {
+                            throw new InvalidOperationException("Unexpected type."); // Sanity check
+                        }
+                        templateObj = (JObject)templateSpecVersion.Template;
+                    }
+                    catch (TemplateSpecsErrorException e)
                     {
-                        throw new InvalidOperationException("Unexpected type."); // Sanity check
+                        //If the templateSpec resourceID is pointing to a non existant resource
+                        if(e.Response.StatusCode.Equals(HttpStatusCode.NotFound))
+                        {
+                            //By returning null, we are introducing parity in the way templateURI and templateSpecId are validated. Gives a cleaner error message in line with the error message for invalid templateURI
+                            return null;
+                        }
+                        //Throw for any other error that is not due to a 404 for the template resource.
+                        throw e;
```

Comment:
Doing a throw of the exception by referencing it loses the stack trace. Just "throw;" (without 'e') should be sufficient here.

Reply:
Pushed the requested change.